ledfx/effects/oneshots/oneshot.py

```diff
@@ -0,0 +1,86 @@
+import timeit
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+
+class Oneshot(ABC):
+    """
+    Base class for all oneshot types.
+
+    Oneshot is an effect that is triggered manually (through, for example, REST API).
+    """
+
+    def __init__(self):
+        self._active: bool = True
+        self._pixel_count: int = 0
+
+    @abstractmethod
```

Comment:
You can probably teach me a lot about proper use of decorators, my python is needs based, rather than educated. Please stick around and throw rocks at my code!

Reply:
Also your use of private and type hinting :-)